user/models.py

```diff
@@ -1,5 +1,48 @@
-from django.contrib.auth.models import AbstractUser
+from typing import Any
+from django.utils.translation import gettext as _
+from django.contrib.auth.models import AbstractUser, BaseUserManager
+from django.db import models
+
+
+class UserManager(BaseUserManager):
+    use_in_migrations = True
+
+    def _create_user(
+            self, email: str, password: str, **extra_fields: dict
+    ) -> "User":
+        if not email:
+            raise ValueError("The given email must be set")
+        email = self.normalize_email(email)
+        user = self.model(email=email, **extra_fields)
+        user.set_password(password)
+        user.save(using=self._db)
+        return user
+
+    def create_user(
+        self, email: str, password: str = None, **extra_fields: Any
```

Comment:
use typehint like: Optional[YOUR_VALUE_TYPE] = None
for example: list that contain string elements and can be None has typehint Optional[list[str]]

**extra_fields: dict